### PR TITLE
More Improvements:

### PR DESCRIPTION
**Pull request by mrbrutti**
- Migrating renderer into a class
- Renderer support of with statement to improve overall flow to build graphs.

### CODE REVIEW

File: exampleTree_complexS3.py [hunk 1]
```diff
@@ -1,9 +1,7 @@
 from models import Action, Block, Detect, Discovery, Edge, Node
-import renderer
+from renderer import Renderer
 
-if __name__ == "__main__":
-    root = Node(label="Reality")
-    goal = Node(label="Attacker gets data from bucket")
+with Renderer(root = "Reality", goal= "Attacker gets data from bucket") as graph:
```
Comment (hyakuhei) [inline exampleTree_complexS3.py:4]: I think we will quickly need graphs with multiple goal-nodes. Not sure about this although we could easily frame it up with goals=["Attacker gets data from bucket"] - what do you think?

File: renderer.py [hunk 2]
```diff
@@ -4,71 +4,87 @@
...
-    if renderUnimplemented == False and unimplemented == True:
-        return
+class Renderer(object):
 
-    if node.__class__.__name__ in dotformat.keys():
-        node_attr = dotformat[node.__class__.__name__]
-        # Overload the default formatting shape if the Node is flagged as unimplemented
-        if unimplemented:
-            node_attr = node_attr | dotformat['_unimplemented_override']
+    def __init__(self, root="Root", goal="Goal"):
```
Comment (hyakuhei) [inline renderer.py:9]: Same comment as above, we likely do want multiple goals

**Review by hyakuhei — APPROVED** [review]

**Comment by hyakuhei** [comment]
This will break any other graphs people have been playing with (and I guess that means we need to quickly get to a stable pattern of models or some versioning or something)

**Comment by mrbrutti** [comment]
Very interesting point on multiple goals I had not thought of that TBH. I guess we could pass a strong or a list (either strings or actual Nodes) and depending on it set one or more elements. Goal1, goal2 or something like that it should not be difficult. Let me think about it and provide you a PoC, but you totally set a great point.

**Pull request merged**